src/com/owncloud/android/ui/preview/PreviewVideoActivity.java

```diff
@@ -1,235 +1,356 @@
-            mSavedPlaybackPosition = extras.getInt(EXTRA_START_POSITION);
-            mAutoplay = extras.getBoolean(EXTRA_AUTOPLAY);
-            
-        } else {
-            mSavedPlaybackPosition = savedInstanceState.getInt(EXTRA_START_POSITION);
-            mAutoplay = savedInstanceState.getBoolean(EXTRA_AUTOPLAY);
-        }
-          
-        mVideoPlayer = (VideoView) findViewById(R.id.videoPlayer);
-
-        // set listeners to get more contol on the playback
-        mVideoPlayer.setOnPreparedListener(this);
-        mVideoPlayer.setOnCompletionListener(this);
-        mVideoPlayer.setOnErrorListener(this);
-          
-        // keep the screen on while the playback is performed (prevents screen off by battery save)
-        mVideoPlayer.setKeepScreenOn(true);
-    }    
-    
-    
-    /**
-     * {@inheritDoc}
-     */
+
+        clearResumePosition();
+
+        setContentView(R.layout.video_preview);
+
+        simpleExoPlayerView = (SimpleExoPlayerView) findViewById(R.id.video_player);
+
+        // Hide sync bar
+        ProgressBar syncProgressBar = (ProgressBar) findViewById(R.id.syncProgressBar);
+        syncProgressBar.setVisibility(View.GONE);
+
+        // Hide full screen button
+        ImageButton fullScreen = (ImageButton) findViewById(R.id.fullscreen_button);
+        fullScreen.setVisibility(View.GONE);
+
+        // Show exit full screen button
+        ImageButton exitFullScreen = (ImageButton) findViewById(R.id.exit_fullscreen_button);
+        exitFullScreen.setVisibility(View.VISIBLE);
+
+        exitFullScreen.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                onBackPressed();
+            }
+        });
+
+
+        Bundle extras = getIntent().getExtras();
+
+        mAutoplay = extras.getBoolean(EXTRA_AUTOPLAY);
+        mPlaybackPosition = extras.getLong(EXTRA_START_POSITION);
+    }
+
     @Override
-    public void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        outState.putInt(PreviewVideoActivity.EXTRA_START_POSITION, mVideoPlayer.getCurrentPosition());
-        outState.putBoolean(PreviewVideoActivity.EXTRA_AUTOPLAY , mVideoPlayer.isPlaying());
+    public void onStart() {
+        super.onStart();
+        Log_OC.v(TAG, "onStart");
+        if (Util.SDK_INT > 23) {
+            preparePlayer();
+        }
     }
 
-    
     @Override
-    public void onBackPressed() {
-        Log_OC.v(TAG, "onBackPressed");
-        Intent i = new Intent();
-        i.putExtra(EXTRA_AUTOPLAY, mVideoPlayer.isPlaying());
-        i.putExtra(EXTRA_START_POSITION, mVideoPlayer.getCurrentPosition());
-        setResult(RESULT_OK, i);
-        mVideoPlayer.stopPlayback();
-        super.onBackPressed();
+    public void onResume() {
+        super.onResume();
+        Log_OC.v(TAG, "onResume");
+        if ((Util.SDK_INT <= 23 || player == null)) {
```

Comment:
I'm not sure about this, I've gotten this from the Exoplayer demo example. At first glance , it has no sense but I would have to check it.